core/src/main/java/io/questdb/griffin/engine/groupby/AbstractSampleByCursor.java

```diff
@@ -103,7 +108,12 @@ protected void parseParams(RecordCursor base, SqlExecutionContext executionConte
             this.nextDstUTC = Long.MAX_VALUE;
         }
 
-        final CharSequence offset = offsetFunc.getStr(null);
+        final CharSequence offset;
+        try {
+            offset = offsetFunc.getStr(null);
+        } catch (UnsupportedOperationException e) {
```

Comment:
`UnsupportedOperationException` is meant to be a placeholder to highlight code inconsistencies in development process. It never meant to be a runtime check.

Perhaps we need to do check upstream somewhere that function is of correct type?

Reply:
Fair point. Addressed in b2928396e75ba80c8cab72bed5358edd86eb1607